tests/slsa_analyzer/test_analyzer.py

```diff
@@ -155,3 +155,18 @@ def test_resolve_analysis_target_no_purl_or_repository() -> None:
     """Test creation of an Analysis Target when no PURL or repository path is provided."""
     with pytest.raises(InvalidAnalysisTargetError):
         Analyzer.to_analysis_target(Configuration(), [], None)
+
+
+@pytest.mark.parametrize(
+    ("url", "purl_string", "expected"),
+    [
+        ("https://github.com:9000/oracle/macaron", "pkg:github/oracle/macaron", True),
+        ("http://user:pass@github.com/oracle/macaron", "pkg:github.com/oracle/macaron", True),
```

Comment:
Are you sure this is a valid URL? I tried passing this through the `parse_remote_url` function in `src/macaron/slsa_analyzer/git_url.py` and it did not parse successfully.

Reply:
It is valid, however, it seems that many browsers have removed support for the feature, or marked it as deprecated. See : [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#access_using_credentials_in_the_url)

The reason it fails with `parse_remote_url` is because that function compares `netloc` against the list of domains, e.g. `github.com`, but `netloc` contains the auth part, the domain, and the port. So most likely that function should be updated to use `hostname` instead.